### DISCUSSION TITLE
Custom request-scoped global proxy with Flask 2.3

### DISCUSSION THREAD

**Discussion by mbarakaja** [pallets/flask#5261] [Q&A] [answered]
I have an application using Flask `2.1.3`. The app has a couple of global variables that proxy values to the request context.

```python
from flask import _request_ctx_stack
from werkzeug.local import LocalProxy

current_org = LocalProxy(
    lambda: getattr(_request_ctx_stack.top, "current_org", None)
)
current_identity = LocalProxy(
    lambda: getattr(_request_ctx_stack.top, "current_identity", None)
)
current_schema = LocalProxy(
    lambda: getattr(_request_ctx_stack.top, "current_schema", None)
)

# This is how values are assigned in a single place
_request_ctx_stack.top.current_org = value
...
```
---

The documentation says we should be using the `g` global proxy, but my application references the global names I posted above, and I would like to avoid a big refactoring.

Is it still possible to have custom global proxies bound to the current request context in Flask 2.3?

**Comment by ThiefMaster** [comment, marked as answer]
Why not simply change your lookup callback to get it from `g` instead of the request ctx stack?

```python
current_org = LocalProxy(lambda: g.get('current_org'))
```

**Reply by mbarakaja** [reply]
Ohh... Thank you so much. This worked for me.
I clearly didn't understand how this worked.
